File: api/src/Diary/DTO/CloudSuggestionDto.php
<?php

namespace App\Diary\DTO;

use Symfony\Component\Uid\UuidV4;

class CloudSuggestionDto
{
    public function __construct(
        public UuidV4 $id,
        /** @var string[] $notes */
        public array $notes,
        public string $suggestion,
        public int $suggestionLinksCount,
        public SuggestionPeriodDto $period,
        public bool $isReceived,
        public ?int $feedbackRating,
        public \DateTimeInterface $createdAt,
    )
    {
    }
}

File: android/app/src/main/java/com/svbackend/natai/android/http/dto/CloudSuggestionDto.kt
package com.svbackend.natai.android.http.dto

import java.time.Instant

data class CloudSuggestionDto(
    val id: String,
    val notes: List<String>,
    val suggestion: String,
    val period: SuggestionPeriodDto,
    val isReceived: Boolean,
    val feedbackRating: Int?,
    val createdAt: Instant,
    val suggestionLinksCount: Int
)